src/js/game/LevelMVC/LevelView.js

```diff
@@ -764,6 +764,7 @@ module.exports = class LevelView {
     } else {
       this.onAnimationEnd(this.playMinecartMoveForwardAnimation(position, facing, isOnBlock, completionHandler, nextPosition, speed), () => {
         this.playTrack(nextPosition, nextFacing, isOnBlock, entity, completionHandler);
+        entity.collectItems(entity.position);
```

Comment:
I'm not sure I love the idea of indiscriminately calling a method that's only defined for the Player entity here, much less the idea of calling a method that was formerly functionally private. Is there somewhere on the Player entity itself we could call this? Perhaps update the player position as it moves around the track, and have that trigger item collection?

Reply:
position is already updated while going around a track so updateMovement() would be a perfectly viable place to put this.

I'm torn on that though, because having a if(onTracks) conditional incorporated into every updateMovement() call, on the one hand, doesn't sound... logical? If we know we want the pick up and track riding to be linked, calling the pick up in the track ride logic makes the most sense to me. 

The easy trap there is, you're accurate, that even though that's 100% safe at the moment (since the only thing that can ride rails is the player) we don't TECHNICALLY know that that will always be the case, so that method isn't very future-proof. If an Agent could ever ride a rail, then we wouldn't want it to pick up miniblocks and then that function call could cause a problem. In the end, I'm not crazy invested either way, so I can move it to updateMovement on Player.js 👍